k8sutil/test_utils.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2016 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class FakeException(Exception):
+    pass
+
+
+class FakeClass(object):
```

Comment:
Do you have a suggestion for a mocking lib? I searched a little and didn't find a good one and found it easier to just write a simple one.

I think in case of this Watch, having mock is not a bad idea, I can break some pieces, but the total behavior (integration test?) should also be tested and that is much easier with a Mock.

Here I didn't focus on pure unit or integration test and tried to cover all code paths as much as possible.